app/src/shared/lib/hooks/useIsToggle.ts

```diff
@@ -0,0 +1,15 @@
+import { useCallback, useState } from 'react';
+
+export const useIsToggle = (initialState?: boolean) => {
+    const [toggle, setToggle] = useState<boolean | undefined>(initialState);
+
+    const handleChangeToggle = useCallback(() => {
+        setToggle((prevState) => !prevState);
+    }, []);
```

Comment:
хук как некая сущность 
описывает свойство(а) и методы работы с ним
поэтому тут лучше 
```suggestion
    const toggle = useCallback(() => {
        setIsToggled((prevState) => !prevState);
    }, []);
```